web/source/dom/domManager.ts

```diff
@@ -238,34 +231,13 @@ namespace com.keyman.dom {
       // Remove any handlers for "NonKMWTouch" elements, since we're enabling it here.
       Pelem.removeEventListener('touchstart', this.nonKMWTouchHandler);
 
-      /*
-      *  Does this element already have a simulated touch element established?  If so,
-      *  just reuse it - if it isn't still in the input list!
-      */
-      if(Pelem['kmw_ip']) {
-
-        if(this.inputList.indexOf(Pelem['kmw_ip']) != -1) {
-          return false;
-        }
-
-        this.inputList.push(Pelem['kmw_ip']);
-
-        console.log("Unexpected state - this element's simulated input DIV should have been removed from the page!");
-
-        return true;   // May need setup elsewhere since it's just been re-added!
-      }
-
-      // The simulated touch element doesn't already exist?  Time to initialize it.
-      let x=dom.constructTouchAlias(Pelem);
-      if(this.isAttached(x)) {
-        x._kmwAttachment.interface = dom.targets.wrapElement(x);
-      } else {
-        this.setupElementAttachment(x); // The touch-alias should have its own wrapper.
+      if(!this.isAttached(Pelem)) {
+        this.setupElementAttachment(Pelem);
+        Pelem.inputMode = 'none';
```

Comment:
Just realised ... we need to make sure hardware keyboards on Android still work with inputMode=none. I guess they will, but assumptions are ...

Reply:
Huh.  Well, I certainly didn't expect _this_:

With `.inputMode = 'none'`, the second I type a character from my laptop's physical keyboard to act as an external keyboard for testing this via Android Studio's device emulation...

![image](https://user-images.githubusercontent.com/25213402/194796737-b9964397-b675-41f2-b913-b917e4537af0.png)

I can confirm that the affected input element's `.inputMode` is still set to `'none'` via Chrome remote-device inspection console.

That might be related to this setting:

![image](https://user-images.githubusercontent.com/25213402/194796948-8617648d-a962-450c-81aa-8b41cd7a6fbc.png)

(I reached that via the bottom-right keyboard-ish looking button beneath the ENTER key.)